### PR TITLE
feat: improve join docs

### PR DESCRIPTION
**Pull request by fuyufjh**
## Description

- [x] Add examples for emit-on-window-close version for window join and interval join
- [x] Correct the example of temporal join.
- [x] Add diagram to illustrate window join and interval join.

## Related code PR

These changes are not motivated by a specific code change but targeted at improving the readability of existing docs.

## Related doc issue

NA

## Rendered preview

https://pr-2586.d2fbku9n2b6wde.amplifyapp.com/docs/next/query-syntax-join-clause/

## Checklist

- [x] I have checked the doc site preview, and the updated parts look good.
- [x] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

File: docs/sql/query-syntax/query-syntax-join-clause.md [hunk 1]
```diff
@@ -134,13 +178,34 @@ FROM s1 JOIN s2
...
-### Append-only process-time temporal join
+```sql
+CREATE MATERIALIZED VIEW interval_join AS
+SELECT s1.id AS id1,
+       s1.value AS value1,
+       s2.id AS id2,
+       s2.value AS value2
+FROM s1 JOIN s2
+ON s1.id = s2.id and s1.ts between s2.ts and s2.ts + INTERVAL '1' MINUTE
+EMIT ON WINDOW CLOSE;
```
Comment (chenzl25) [inline docs/sql/query-syntax/query-syntax-join-clause.md]: EMIT ON WINDOW CLOSE is unsupported for joins. We shouldn't use it as an example.

File: docs/sql/query-syntax/query-syntax-join-clause.md [hunk 2]
```diff
@@ -194,34 +265,29 @@ ON product_id = id WHERE process_time BETWEEN valid_from AND valid_to;
 | 2              | 102        | 2        | 2023-06-19 | Product B    | 15    |
 | 3              | 101        | 1        | 2023-06-20 | Product A    | 22    |
 
-### Non-append-only process-time temporal join
-
-Compared to the append-only temporal join, the non-append-only temporal join can accommodate non-append-only input for the left table. However, it introduces an internal state to materialize the lookup result for each left-hand side (LHS) insertion. This allows the temporal join operator to retract the join result it sends downstream when update or delete messages arrive.
-
-#### Syntax
+Notice that the Produce A's price is 25 at the time of transaction 1 and 22 at the time of transaction 3.
```
Comment (WanYixian) [inline docs/sql/query-syntax/query-syntax-join-clause.md]:
```suggestion
Notice that the Product A's price is 25 at the time of transaction 1 and 22 at the time of transaction 3.
```

File: docs/sql/query-syntax/query-syntax-join-clause.md [hunk 3]
```diff
@@ -110,6 +115,42 @@ JOIN TUMBLE(s2, ts, interval '1' MINUTE)
...
+EMIT ON WINDOW CLOSE;
+```
+This query will join the two sources `s1` and `s2` based on the `id` column and the window start time. The join result will be outputted only when the window closes, so unclosed windows will not be shown in the materialized view or downstream.
+
+See [Emit on Window Close](transform/emit-on-window-close.md) for more information on the `EMIT ON WINDOW CLOSE` clause.
+
+</TabItem>
+</Tabs>
+
+### State Cleaning
```
Comment (WanYixian) [inline docs/sql/query-syntax/query-syntax-join-clause.md]:
```suggestion
### State cleaning
```

File: docs/sql/query-syntax/query-syntax-join-clause.md [hunk 4]
```diff
@@ -134,13 +178,34 @@ FROM s1 JOIN s2
 ON s1.id = s2.id and s1.ts between s2.ts and s2.ts + INTERVAL '1' MINUTE;
 ```
 
-## Process-time temporal joins
+This query will join the two sources `s1` and `s2` based on the `id` column and the time range. For the unclosed windows, the join result will be updated immediately when new data arrives.
```
Comment (WanYixian) [inline docs/sql/query-syntax/query-syntax-join-clause.md]:
```suggestion
This query will join the two sources `s1` and `s2` based on the `id` column and the time range. For the unclosed windows, the join result will be updated immediately when new data arrive.
```

**Review by WanYixian — APPROVED** [review]
LGTM!

File: docs/sql/query-syntax/query-syntax-join-clause.md (hunk 1 above)
Comment (fuyufjh) [reply docs/sql/query-syntax/query-syntax-join-clause.md]:
```
dev=> CREATE SOURCE s1 (
 id int,
 value int,
 ts TIMESTAMP,
 WATERMARK FOR ts AS ts - INTERVAL '20' SECOND
) WITH (
    connector = 'datagen',
    fields.ts.kind = 'random',
    fields.ts.max_past = '20 seconds',
    fields.ts.max_past_mode = 'relative',
    datagen.rows.per.second='1',
);

CREATE SOURCE s2 (
 id int,
 value int,
 ts TIMESTAMP,
 WATERMARK FOR ts AS ts - INTERVAL '20' SECOND
) WITH (
    connector = 'datagen',
    fields.ts.kind = 'random',
    fields.ts.max_past = '20 seconds',
    fields.ts.max_past_mode = 'relative',
    datagen.rows.per.second='1',
);
CREATE_SOURCE
CREATE_SOURCE
dev=> CREATE MATERIALIZED VIEW interval_join AS
SELECT s1.id AS id1,
       s1.value AS value1,
       s2.id AS id2,
       s2.value AS value2
FROM s1 JOIN s2
ON s1.id = s2.id and s1.ts between s2.ts and s2.ts + INTERVAL '1' MINUTE
EMIT ON WINDOW CLOSE;
NOTICE:  EMIT ON WINDOW CLOSE is currently an experimental feature. Please use it with caution.
ERROR:  Failed to run the query

Caused by:
  Not supported: The query cannot be executed in Emit-On-Window-Close mode.
HINT: Try define a watermark column in the source, or avoid aggregation without GROUP BY
```

Indeed...

Comment (fuyufjh) [reply docs/sql/query-syntax/query-syntax-join-clause.md]: https://github.com/risingwavelabs/risingwave/issues/18445

**Comment by fuyufjh** [comment]
Have added illustrations and resolved comments. Please take a look again.

**Review by st1page — APPROVED** [review]

**Pull request merged**